Rubberduck.Core/UI/Command/CommandBase.cs

```diff
@@ -22,16 +22,42 @@ protected CommandBase(ILogger logger = null)
         protected abstract void OnExecute(object parameter);
 
         protected Func<object, bool> CanExecuteCondition { get; private set; }
+        protected Func<object, bool> OnExecuteCondition { get; private set; }
+        private bool RequireReEvaluationOnExecute => OnExecuteCondition != null;
 
-        protected void AddToCanExecuteEvaluation(Func<object, bool> furtherCanExecuteEvaluation)
+        protected void AddToCanExecuteEvaluation(Func<object, bool> furtherCanExecuteEvaluation, bool requireReevaluation = false)
         {
             if (furtherCanExecuteEvaluation == null)
             {
                 return;
             }
 
-            var currentCanExecute = CanExecuteCondition;
-            CanExecuteCondition = (parameter) => currentCanExecute(parameter) && furtherCanExecuteEvaluation(parameter);
+            AddToCanExecuteEvaluation(furtherCanExecuteEvaluation);
+
+            if (requireReevaluation)
+            {
+                AddToOnExecuteEvaluation(furtherCanExecuteEvaluation);
+            }
+        }
+
+        private void AddToCanExecuteEvaluation(Func<object, bool> furtherCanExecuteEvaluation)
+        {
+            var currentCanExecuteCondition = CanExecuteCondition;
+            CanExecuteCondition = (parameter) =>
+                currentCanExecuteCondition(parameter) && furtherCanExecuteEvaluation(parameter);
+        }
+
+        private void AddToOnExecuteEvaluation(Func<object, bool> furtherCanExecuteEvaluation)
+        {
+            if (OnExecuteCondition == null)
```

Comment:
Why not simply set `OnExecuteEvaluation` to `() => true` in the constructor like for the `CanExecuteEvaluation`. That removes the need for all those `null` checks.